data/reusables/github-actions/self-hosted-runner-labels-runs-on.md

```diff
@@ -1,3 +1,5 @@
 To specify a self-hosted runner for your job, configure `runs-on` in your workflow file with self-hosted runner labels.
 
-All self-hosted runners have the `self-hosted` label. Using only this label will select any self-hosted runner. To select runners that meet certain criteria, such as operating system or architecture, provide an array of labels that begins with `self-hosted` (this must be listed first) and then includes additional labels as needed.
+All self-hosted runners have the `self-hosted` label. Using only this label will select any self-hosted runner. To select runners that meet certain criteria, such as operating system or architecture, we recommend providing an array of labels that begins with `self-hosted` (this must be listed first) and then includes additional labels as needed.
+
+Although the `self-hosted` label is not required, we strongly recommend specifying it when using self-hosted runners to ensure that your job does not unintentionally specify any current or future {% data variables.product.prodname_dotcom %}-hosted runners.
```

Comment:
🤔 I'm hesitant to describe that in the docs, because it might mislead users to thinking that self-hosted runners are always preferred if they use the same name as a GitHub-hosted one. IIUC, it only prefers the self-hosted runner if it is online and idle. e.g. if the self-hosted runner is busy processing jobs and there are GitHub-hosted runners available, then the job will be routed to a GitHub-hosted runner. The only fool-proof way to make sure that it'll be run on the self-hosted runner is to include the `self-hosted` label.

@ethomson Is my understanding of ☝️ correct?

Reply:
That's my understanding as well.